src/main/msp/msp_serial.c

```diff
@@ -120,40 +196,84 @@ static uint8_t mspSerialChecksumBuf(uint8_t checksum, const uint8_t *data, int l
     return checksum;
 }
 
+static uint8_t mspSerialChecksumBufV2(uint8_t checksum, const uint8_t *data, int len)
+{
+    while (len-- > 0) {
+        checksum = crc8_dvb_s2(checksum, *data++);
+    }
+    return checksum;
+}
+
 #define JUMBO_FRAME_SIZE_LIMIT 255
 
-static int mspSerialEncode(mspPort_t *msp, mspPacket_t *packet)
+static int mspSerialEncode(mspPort_t *msp, mspPacket_t *packet, bool isMSPv2)
```

Comment:
Using `bool` as parameter type makes code hard to read. Instead create enum with values MSP_V1 and MSP_V2.